util-taglib/src/com/liferay/taglib/ui/ErrorTag.java

```diff
@@ -354,6 +355,20 @@ private boolean _isShowAlert() {
 		}
 
 		if (_key == null) {
+			if (!_embed) {
+				boolean suppressDefaultToastMessage = SessionErrors.contains(
+					httpServletRequest,
+					"liferay-ui:error:suppressDefaultToastMessage");
+
+				if (suppressDefaultToastMessage) {
+					return false;
+				}
+
+				SessionErrors.add(
+					httpServletRequest,
+					"liferay-ui:error:suppressDefaultToastMessage");
+			}
+
```

Comment:
@holatuwol This does not seem right. We are adding adding logic for a specific error to a general error class. The issue should ideally be fixed on portlet level, not on tag level.

An error thrown by a portlet should normally be handled only by that portlet. Why is this case different? Can we apply standard error behavior?

Reply:
This code adds logic that only runs for non-specific errors (no key specified).

Reply:
Let me clarify my question. I am not referring to code in this PR, I am referring to error-handling code in site navigation menu and search bar. You said in the opening comment they are the ones throwing errors. Why are they displaying errors thrown by a different portlet? This is not standard behavior. Why is this case different? Can we apply standard error behavior in those two portlets?

Reply:
The gist of it is, we can't actually know which portlet generated the error message because it wasn't a portlet that generated the error, but rather a password reset struts action that's not tied to any portlet.

This is effectively being done through `portlet_messages/page.jsp` which is run for every portlet (so it's not as though these portlets are specifically showing all errors, but rather than portlet_messages, which is run for every portlet, will show the errors).

The issue happens due to changes made in [LPS-99142](https://issues.liferay.com/browse/LPS-99142), and I can fix it by adding another check to only render that message if it's the current portlet, but then the fix is functionally equivalent to calling SessionErrors instead of MultiSessionErrors, which makes me wonder if this change introduces some kind of regression that I can't understand just by looking at the code.

As part of due diligence, I test and confirmed that the original issue reported in LPS-99142 does not recur with the new changes, but I don't know whether there is another edge case that I should test. This is why I feel like an ErrorTag check would be safer, but I've applied the changes to `portlet_messages/page.jsp` instead.